core/trino-main/src/main/java/io/trino/operator/OutputSpoolingOperatorFactory.java

```diff
@@ -316,25 +315,37 @@ private static Block[] emptyBlocks(Map<Symbol, Integer> layout)
 
             return blocks;
         }
+
+        @Override
+        public void close()
+                throws Exception
+        {
+            userMemoryContext.close();
+        }
```

Comment:
I do not think it needs explicit closing. At least we do not do that elsewhere

Reply:
It is by using closer

Reply:
Close is called in multiple places around the code.

Reply:
Actually there is a comment that I saw:

```
// caller should close this context as it's a new context
    public LocalMemoryContext newLocalUserMemoryContext(String allocationTag)
    {
        return new InternalLocalMemoryContext(operatorMemoryContext.newUserMemoryContext(allocationTag), memoryFuture, this::updatePeakMemoryReservations, true);
    }
```